presto-cassandra/src/test/java/com/facebook/presto/cassandra/TestCassandraIntegrationSmokeTest.java

```diff
@@ -13,34 +13,195 @@
+        assertSelect(TABLE_ALL_TYPES_PARTITION_KEY, false);
+    }
 
-        queryRunner.execute(createCassandraSession(keyspace), "select * from presto_test where key='key 1'");
-        queryRunner.execute(createCassandraSession(keyspace), "select * from presto_test where key='key 2'");
-        queryRunner.execute(createCassandraSession(keyspace), "select * from presto_test where key='key 3'");
+    @Test
+    public void testCreateTableAs()
+            throws Exception
+    {
+        execute("DROP TABLE IF EXISTS table_all_types_copy");
+        execute("CREATE TABLE table_all_types_copy AS SELECT * FROM " + TABLE_ALL_TYPES);
+        assertSelect("table_all_types_copy", true);
+        execute("DROP TABLE table_all_types_copy");
+    }
+
+    private void assertSelect(String tableName, boolean createdByPresto)
+    {
+        Type uuidType = createdByPresto ? createUnboundedVarcharType() : createVarcharType(36);
+        Type inetType = createdByPresto ? createUnboundedVarcharType() : createVarcharType(45);
+
+        String sql = "SELECT " +
+                " key, " +
+                " typeuuid, " +
+                " typeinteger, " +
+                " typelong, " +
+                " typebytes, " +
+                " typetimestamp, " +
+                " typeansi, " +
+                " typeboolean, " +
+                " typedecimal, " +
+                " typedouble, " +
+                " typefloat, " +
+                " typeinet, " +
+                " typevarchar, " +
+                " typevarint, " +
+                " typetimeuuid, " +
+                " typelist, " +
+                " typemap, " +
+                " typeset " +
+                " FROM " + tableName;
+
+        MaterializedResult result = execute(sql);
+
+        int rowCount = result.getRowCount();
+        assertEquals(rowCount, 9);
+        assertEquals(result.getTypes(), ImmutableList.of(
+                createUnboundedVarcharType(),
+                uuidType,
+                INTEGER,
+                BIGINT,
+                VARBINARY,
+                TIMESTAMP,
+                createUnboundedVarcharType(),
+                BOOLEAN,
+                DOUBLE,
+                DOUBLE,
+                DOUBLE,
+                inetType,
+                createUnboundedVarcharType(),
+                createUnboundedVarcharType(),
+                uuidType,
+                createUnboundedVarcharType(),
+                createUnboundedVarcharType(),
+                createUnboundedVarcharType()
+        ));
+
+        List<MaterializedRow> sortedRows = result.getMaterializedRows().stream()
+                .sorted((o1, o2) -> o1.getField(1).toString().compareTo(o2.getField(1).toString()))
+                .collect(toList());
+
+        for (int rowNumber = 1; rowNumber <= rowCount; rowNumber++) {
+            assertEquals(sortedRows.get(rowNumber - 1), new MaterializedRow(DEFAULT_PRECISION,
+                    "key " + rowNumber,
```

Comment:
Can you get the values from the util class where the tables are defined? Ideally via an extracted method, but that may not be possible b/c of different types.